app/scripts/transaction-manager.js

```diff
@@ -0,0 +1,207 @@
+  }
+
+  get pendingTxCount () {
+    return this.getTxsByMetaData('status', 'signed').length
+  }
+
+  getUnapprovedTxList () {
+    var txList = this.getTxList()
+    return txList.filter((tx) => {
+      return tx.status === 'unapproved'
+    }).reduce((result, tx) => {
+      result[tx.id] = tx
+      return result
+    }, {})
+  }
+
+  /*
+  Takes an object of fields to search for eg:
+  var thingsToLookFor = {
+    to: '0x0..',
+    from: '0x0..',
+    status: 'signed',
+  }
+  and returns a list of tx with all
+  options matching
+
+  this is for things like filtering a the tx list
+  for only tx's from 1 account
+  or for filltering for all txs from one account
+  and that have been 'confirmed'
+  */
+  getFilterdTxList (opts) {
+    var filteredTxList
+    Object.keys(opts).forEach((key) => {
+      filteredTxList = this.getTxsByMetaData(key, opts[key], filteredTxList)
+    })
+    return filteredTxList
+  }
+
+  getTxsByMetaData (key, value, txList = this.getTxList()) {
+    return txList.filter((tx) => {
+      if (key in tx.txParams) {
+        return tx.txParams[key] === value
+      } else {
+        return tx[key] === value
+      }
+    })
+  }
+
+// keeps around the txCbs for later
+  addOnTxDoneCb (txId, cb) {
+    this._unconfTxCbs[txId] = cb || noop
+  }
+
+  execOnTxDoneCb (txId, conf) {
+    var approvalCb = this._unconfTxCbs[txId]
+
+    approvalCb(null, conf)
+    // clean up
+    delete this._unconfTxCbs[txId]
+  }
+
+  //   should return the tx
+
+  //   Should find the tx in the tx list and
+  //   update it.
+  //   should set the status in txData
+  //     // - `'unapproved'` the user has not responded
+  //     // - `'rejected'` the user has responded no!
+  //     // - `'signed'` the tx is signed
+  //     // - `'submitted'` the tx is sent to a server
+  //     // - `'confirmed'` the tx has been included in a block.
+  setTxStatus (txId, status) {
+    var txData = this.getTx(txId)
+    txData.status = status
+    this.emit(status, txId)
+    this.updateTx(txData, status)
+  }
+
+
+  //   should return the status of the tx.
+  getTxStatus (txId, cb) {
+    const txData = this.getTx(txId)
+    return cb ? cb(txData.staus) : txData.status
+  }
+
+
+  //   should update the status of the tx to 'signed'.
+  setTxStatusSigned (txId) {
+    this.setTxStatus(txId, 'signed')
+    this.emit('update')
+  }
+
+  //     should update the status of the tx to 'rejected'.
+  setTxStatusRejected (txId) {
+    this.setTxStatus(txId, 'rejected')
+    this.emit('update')
+  }
+
+  setTxStatusConfirmed (txId) {
+    this.setTxStatus(txId, 'confirmed')
+  }
+
+  // merges txParams obj onto txData.txParams
+  // use extend to ensure that all fields are filled
+  updateTxParams (txId, txParams) {
+    var txData = this.getTx(txId)
+    txData.txParams = extend(txData, txParams)
+    this.updateTx(txData)
+  }
+
+  //  sets provider for provider utils and event listener
+  setProvider (provider) {
```

Comment:
in the off-chance that there is an existing provider, we should clean up old listeners.

This could also be solved by moving this all into the constructor. (happens once)